integrationtest/integration_test.go

```diff
@@ -51,29 +74,97 @@ func TestUnauthorizedServiceAccountErrorCode(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	defer func() {
+	deferred := func() {
 		_, err = client.Logical().Delete("sys/auth/kubernetes")
 		if err != nil {
 			t.Fatal(err)
 		}
+	}
+
+	defer func() {
+		// just in case setupKubernetesAuth panics before returning deferred to the caller
+		if panicErr := recover(); panicErr != nil {
+			deferred()
+			panic(panicErr)
+		}
```

Comment:
It's a little trickier -- if this `setupKubernetesAuth` function fails, we want to cleanup (i.e., `DELETE sys/auth/kubernetes`).

But, we can't do it in a normal `defer` because that would be called when the function returns. We want to call the `deferred` later, so we pass it back so that the caller can `defer` it, and the resource won't disappear before they use it.

Reply:
Right, it makes sense to return `deferred` so that it can be called later. However, I think that we could still conditionally call `deferred` in the case of failure from this `setupKubernetesAuth` itself without the use of `recover`. IOW _this_ defer could be:
```
defer func() {
		// just in case setupKubernetesAuth errors before returning deferred to the caller
		if t.Failed() {
			deferred()
		}
}()
```

Wouldn't we also want to perform clean up if this function errors (e.g. from the config operation below), and not just on panics?

Reply:
Ah, I see. You're right. I'll also execute it on `t.Failed()` -- I think we should keep the `panic` check as well.